common/hooks/useAuth.js

```diff
@@ -145,17 +145,22 @@ const useAuth = () => {
     const verifier = window.localStorage.getItem('auth.verifier');
     const token = await getToken(code, verifier);
 
-    window.localStorage.setItem('auth.token', JSON.stringify(token));
-    setState({ type: authStates.authorized, token });
-    const link = {
-      pathname: Router.asPath.split('?')[0],
-      // TODO: This is very app specific, but it's an absolute pain to try and get this to work
-      // The side effect is that it removes the rest of the URL params if there are any
-      query: {
-        action: Router.query.action,
-      },
-    };
-    Router.replace(link, link, { shallow: true });
+    if (token) {
+      window.localStorage.setItem('auth.token', JSON.stringify(token));
+      setState({ type: authStates.authorized, token });
+      const link = {
+        pathname: Router.asPath.split('?')[0],
+        // TODO: This is very app specific, but it's an absolute pain to try and get this to work
+        // The side effect is that it removes the rest of the URL params if there are any
+        query: {
+          action: Router.query.action,
+        },
+      };
+      Router.replace(link, link, { shallow: true });
+    } else {
+      console.error('getToken returned undefined. Failed to authorise.');
```

Comment:
Should we not take them back to login?

Reply:
yeah, probably. Do you think it's worth having an 'authorisationFailed' state as well, so we can indicate that there was a problem and the user should try again?

Reply:
Would it be similar to the auth needing refreshing?

Reply:
This state would also probably not be exposed to the component as we should just treat it as not logged in.

Reply:
Is there a way to do that while indicating that there was a problem to the user somehow?